x/posts/internal/types/post_media.go

```diff
@@ -0,0 +1,120 @@
+package types
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// ---------------
+// --- PostMedia
+// ---------------
+
+type PostMedia struct {
+	URI      string `json:"uri"`
+	MimeType string `json:"mime_Type"`
+}
+
+func NewPostMedia(uri, mimeType string) PostMedia {
+	return PostMedia{
+		URI:      uri,
+		MimeType: mimeType,
+	}
+}
+
+// String implements fmt.Stringer
+func (pm PostMedia) String() string {
+	out := "Media - "
+
+	out += fmt.Sprintf(" URI - [%s] ; Mime-Type - [%s] \n", pm.URI, pm.MimeType)
+
+	return strings.TrimSpace(out)
+}
+
+// Validate implements validator
+func (pm PostMedia) Validate() error {
+	if len(strings.TrimSpace(pm.URI)) == 0 {
+		return fmt.Errorf("uri must be specified and cannot be empty")
+	}
+
+	if err := ParseURI(pm.URI); err != nil {
+		return err
+	}
+
+	if len(strings.TrimSpace(pm.MimeType)) == 0 {
+		return fmt.Errorf("mime type must be specified and cannot be empty")
+	}
+
+	return nil
+}
+
+// Equals allows to check whether the contents of pm are the same of other
+func (pm PostMedia) Equals(other PostMedia) bool {
+	return pm.URI == other.URI && pm.MimeType == other.MimeType
+}
+
+// ParseURI checks if the given uri string is well-formed according to the regExp and return and error otherwise
+func ParseURI(uri string) error {
```

Comment:
I would advise renaming this method to something more like `ValidateURI` or `IsURIValid` (in the latter case, returning a `bool` instead of an `error). 

Usually we refer to methods called `ParseXXX` to the ones that actually return a new type from a string, which is not done here.